stdlib/public/core/HashedCollections.swift.gyb

```diff
@@ -2473,6 +2473,11 @@ collections = [
 /// keys, followed by the values.
 //
 // See the docs at the top of the file for more details on this type
+//
+// NOTE: The precise layout of this type is relied on in the runtime
+// to provide a statically allocated empty singleton. 
```

Comment:
No, fixed-layout just opts into a layout algorithm (some day) guaranteed to be consistent across module boundaries and across Swift compiler versions. If you want to actually control layout, you have to actually define the type in C for the foreseeable future.